[svg] applying `rx` or `ry` through CSS exclusively has no effect

graouts opened the pull request:
#### 0eba4b6cab1736dee19afb4397af66a46b9cb01b
<pre>
[svg] applying `rx` or `ry` through CSS exclusively has no effect
<a href="https://bugs.webkit.org/show_bug.cgi?id=259646">https://bugs.webkit.org/show_bug.cgi?id=259646</a>
<a href="https://rdar.apple.com/113500023">rdar://113500023</a>

Reviewed by Said Abou-Hallawa.

We should not be checking the XML attribute values for `rx` and `ry`
to determine whether a &lt;rect&gt; is rounded, but instead go through the
`SVGRenderStyle` which will the attribute values reflected as well as
account for the CSS properties.

* LayoutTests/imported/w3c/web-platform-tests/svg/shapes/rect-rx-set-by-css-expected.svg: Added.
* LayoutTests/imported/w3c/web-platform-tests/svg/shapes/rect-rx-set-by-css-ref.svg: Added.
* LayoutTests/imported/w3c/web-platform-tests/svg/shapes/rect-rx-set-by-css.svg: Added.
* Source/WebCore/rendering/svg/RenderSVGRect.cpp:
(WebCore::RenderSVGRect::updateShapeFromElement):
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGRect.cpp:
(WebCore::LegacyRenderSVGRect::updateShapeFromElement):

Canonical link: <a href="https://commits.webkit.org/271970@main">https://commits.webkit.org/271970@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8fddb64cb83871df401f19515baf168eeb559f77

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/30224 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/8893 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/31690 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/32729 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/27341 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/30906 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/11093 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/6130 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/27325 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/30528 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/7469 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/26249 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/6390 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/6541 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/26866 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/34069 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/27542 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/27301 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/32720 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/6496 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/4662 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/30534 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/8246 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/7251 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3899 "Built successfully and passed tests") | [❌ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/7024 "Failed to checkout and rebase branch from PR 21679") | | | 
<!--EWS-Status-Bubble-End-->

shallawa reviewed (approved):
The title of the PR includes `CSS animation` although the bug can be reproduced without any animation. Also the layout test includes an SVG static rounded rect. Can you please change the title or add a new animation test?

graouts committed:
[svg] applying `rx` or `ry` through CSS exclusively has no effect
https://bugs.webkit.org/show_bug.cgi?id=259646
rdar://113500023

Reviewed by Said Abou-Hallawa.

We should not be checking the XML attribute values for `rx` and `ry`
to determine whether a <rect> is rounded, but instead go through the
`SVGRenderStyle` which will the attribute values reflected as well as
account for the CSS properties.

* LayoutTests/imported/w3c/web-platform-tests/svg/shapes/rect-rx-set-by-css-expected.svg: Added.
* LayoutTests/imported/w3c/web-platform-tests/svg/shapes/rect-rx-set-by-css-ref.svg: Added.
* LayoutTests/imported/w3c/web-platform-tests/svg/shapes/rect-rx-set-by-css.svg: Added.
* Source/WebCore/rendering/svg/RenderSVGRect.cpp:
(WebCore::RenderSVGRect::updateShapeFromElement):
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGRect.cpp:
(WebCore::LegacyRenderSVGRect::updateShapeFromElement):

Canonical link: https://commits.webkit.org/271970@main
